mdagent/tools/base_tools/analysis_tools/secondary_structure.py

```diff
@@ -0,0 +1,551 @@
+                traj_fileid=traj_file,
+                top_fileid=top_file,
+            )
+            if not traj:
+                raise Exception("Trajectory could not be loaded.")
+        except Exception as e:
+            return str(e)
+        relative_shape_antisotropy_values = self._compute_relative_shape_antisotropy(
+            traj
+        )
+        if traj.n_frames == 1:
+            return (
+                "Relative shape antisotropy computed for "
+                "a single frame, no file saved."
+                f"Relative shape antisotropy: {relative_shape_antisotropy_values}"
+            )
+
+        raw_file_id = write_raw_x(
+            "relative_shape_antisotropy",
+            relative_shape_antisotropy_values,
+            traj_file,
+            self.path_registry,
+        )
+        plot_file_id = plot_x_over_time(
+            "relative_shape_antisotropy",
+            relative_shape_antisotropy_values,
+            traj_file,
+            self.path_registry,
+        )
+        return (
+            "relative_shape_antisotropy_values saved to "
+            f"{raw_file_id}, plot saved to "
+            f"{plot_file_id}"
+        )
+
+    async def _arun(self, traj_file, top_file):
+        """Runs the tool asynchronously."""
+        raise NotImplementedError("Async version not implemented")
+
+
+class AnalyzeProteinStructure(BaseTool):
+    name = "AnalyzeProtein"
+    description = (
+        "Analyze a protein trajectory. "
+        "Input is a trajectory file ID"
+        "and an optional topology file ID, "
+        "along with a list of analyses to perform."
+        "enter the analyses you want to perform as "
+        "a string, separated by commas. "
+        "The output is a dictionary "
+        "containing the requested analyses."
+    )
+    path_registry: PathRegistry = PathRegistry.get_instance()
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def analyze_protein(self, traj, requested_analyses: list):
+        result = {}
+        if "n_atoms" in requested_analyses:
+            result["n_atoms"] = traj.n_atoms
+        if "n_residues" in requested_analyses:
+            result["n_residues"] = traj.n_residues
+        if "n_chains" in requested_analyses:
+            result["n_chains"] = traj.n_chains
+        if "n_frames" in requested_analyses:
+            result["n_frames"] = traj.n_frames
+        if "time" in requested_analyses:
+            result["time"] = traj.time
+        if "time_step" in requested_analyses:
+            result["time_step"] = traj.time_step
+        if "atoms" in requested_analyses:
+            result["atoms"] = traj.topology.atoms
+        if "bonds" in requested_analyses:
+            result["bonds"] = traj.topology.bonds
+        if "chains" in requested_analyses:
+            result["chains"] = traj.topology.chains
+        if "residues" in requested_analyses:
```

Comment:
do we want the total list of residues? or the list of unique residues? the total list of residues can be veery extent, and of atoms, even longer. So im not sure if is a good idea to return all of them. Maybe a summary of the residues would make more sense like: {"ALA": total count 35....}. with atoms, even less sure.

Now that im looking at the documentation, this tool won't work. 
atoms, residues, chains, and bonds are generators, so they can be used in a for loop, but not to see the list as strings. 

```
print(str(traj.topology.residues))
>> <generator object Topology.residues at ....> 
```

Here is some code that I was experimenting on:
```
new_topology = pd.DataFrame(
    {'serial':range(traj.n_residues),
    'name':['COM' for _ in traj.topology.residues],
    'resSeq':range(traj.n_residues),
    'resName':[i.name for i in traj.topology.residues],
    'element':["VS" for _ in traj.topology.residues],
    'chainID':[i.chain.chain_id if i.chain.chain_id else 0 for i in traj.topology.residues],
    'segmentID':[i.segment_id for i in traj.topology.residues]
    }
)
print(new_topology.head())
>>  serial name  resSeq resName element  chainID segmentID
0       0  COM       0     MET      VS        0      4AKE
1       1  COM       1     ARG      VS        0      4AKE
2       2  COM       2     ILE      VS        0      4AKE
3       3  COM       3     ILE      VS        0      4AKE
4       4  COM       4     LEU      VS        0      4AKE
```

Not that this is what you wanted, but to show how to work with the generators :) Again, instead of huge lists, a summary of the number of residues, atoms, and their ids per chain would be nicer?

Reply:
lol - you're right

I think this tool really just makes sense to count, I dont think there is much value otherwise